### PR TITLE
Development to Master. Istanbul changes. buidl.

### PR DESCRIPTION
**Pull request by ilanDoron**

### CODE REVIEW

File: contracts/KyberNetwork.sol [hunk 1]
```diff
@@ -419,6 +418,27 @@ contract KyberNetwork is Withdrawable, Utils2, KyberNetworkInterface, Reentrancy
     }
     /* solhint-enable code-complexity */
 
+    function getReservesRates(ERC20 token, uint optionalAmount) public view
+        returns(address[] buyReserves, uint[] buyRates, address[] sellReserves, uint[] sellRates)
+    {
+        uint amount = optionalAmount > 0 ? optionalAmount : 1000;
```
[the remark this answers is not in the document]
Comment (ilanDoron) [reply contracts/KyberNetwork.sol]: If 1000 is too big. the caller will have to modify the amount.

but as a general case 1000 seems safe enough

File: contracts/reserves/fprConversionRate/EnhancedStepFunctions.sol [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+        int[],
+        int[]
+    )
+        public
+        onlyOperator
+    {
+        revert();
+    }
+
+    function setImbalanceStepFunction(
```
[the remark this answers is not in the document]
Comment (ilanDoron) [reply contracts/reserves/fprConversionRate/EnhancedStepFunctions.sol]: lets do this in a separate PR

File: contractsV5/V5Example.sol [hunk 3]
```diff
@@ -0,0 +1,18 @@
+pragma solidity 0.5.11;
```
[the remark this answers is not in the document]
Comment (ilanDoron) [reply contractsV5/V5Example.sol]: lets remove it after we PR the real V5 files in your eth2Dai PR

File: test/v5Example.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+const Helper = require("./helper.js");
```
[the remark this answers is not in the document]
Comment (ilanDoron) [reply test/v5Example.js]: OK
lets remove after Eth2Dai is merged to master

File: contractsV5/V5Example.sol (hunk 3 above)
[the remark this answers is not in the document]
Comment (Anyhowclick) [reply contractsV5/V5Example.sol]: Yup, can be removed after Mike's PR.

File: web3deployment/basicDeployer.js [hunk 5]
```diff
@@ -18,9 +18,7 @@ const web3 = new Web3(new Web3.providers.HttpProvider(rpcUrl));
 const solc = require('solc')
 
 const rand = web3.utils.randomHex(7);
-
 const privateKey = web3.utils.sha3("in joy we trust" + rand);
```
Comment (Anyhowclick) [inline web3deployment/basicDeployer.js]: Lol what.... Anton wrote this deployer?

Comment (ilanDoron) [reply web3deployment/basicDeployer.js]: HAHA
no.
Actually it was originally "js sucks"
I wanted something more optimistic and it ended up like this...

**Pull request merged**